packages/flutter/lib/src/material/button.dart

```diff
@@ -278,14 +292,18 @@ class _RawMaterialButtonState extends State<RawMaterialButton> {
 
   @override
   Widget build(BuildContext context) {
+    final Color effectiveTextColor = MaterialStateColor.getColorInStates(widget.textStyle?.color, _states);
+
     final Widget result = Focus(
       focusNode: widget.focusNode,
-      onFocusChange: (bool focused) => setState(() { _focused = focused; }),
+      onFocusChange: (bool focused) => setState(() {
+        _updateState(focused, MaterialState.focused);
+      }),
       child: ConstrainedBox(
         constraints: widget.constraints,
         child: Material(
           elevation: _effectiveElevation(),
-          textStyle: widget.textStyle,
+          textStyle: widget?.textStyle?.copyWith(color: effectiveTextColor),
```

Comment:
The widget can't be null so you don't need the first `?`

Reply:
Yep, thanks for spotting that.